### PR TITLE
Add examples for swc110

### PR DESCRIPTION
**Pull request by b-mueller**

### CODE REVIEW

File: test_cases/assert_violations/assert_constructor.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+description: "Assert violation in constructor"
+issues:
+- id: "SWC-110"
+  count: 1
+  locations:
+  - bytecode_offsets: [24]
+    line_numbers: [5]
```
Comment (thec00n) [inline test_cases/assert_violations/assert_constructor.yaml]: 5?

Comment (b-mueller) [reply test_cases/assert_violations/assert_constructor.yaml]: 10.

File: test_cases/assert_violations/out-of-bounds-exception.sol [hunk 2]
```diff
@@ -0,0 +1,11 @@
+pragma solidity ^0.4.19;
+
+contract OutOfBoundsException {
+
+	uint256[] public array;
+
+	function getArrayElement(uint256 idx) returns (uint256) {
+		return array[idx];
```
Comment (thec00n) [inline test_cases/assert_violations/out-of-bounds-exception.sol]: This could throw.

Comment (b-mueller) [reply test_cases/assert_violations/out-of-bounds-exception.sol]: This is meant for an example that could throw but should *not* be reported as an assert violation. Does that make sense?

Comment (thec00n) [reply test_cases/assert_violations/out-of-bounds-exception.sol]: Is this sample related to another variant around exception handling -> https://cwe.mitre.org/data/definitions/703.html ?

Comment (thec00n) [reply test_cases/assert_violations/out-of-bounds-exception.sol]: Also it be great to have the definition SWC-110 completed :)

Comment (b-mueller) [reply test_cases/assert_violations/out-of-bounds-exception.sol]: Added in #48

**Pull request merged**